macros/incremental_utils.sql

```diff
@@ -6,3 +6,11 @@
       true
     {% endif %}
 {%- endmacro %}
+
+{% macro incremental_last_x_days(time_col, time_in_days) -%}
+    {% if is_incremental() %}
+        {{ time_col }} >= current_date() - {{ time_in_days }}
```

Comment:
i'd recommend still using the max timestamp from the table so 
```suggestion
        {{ time_col }} >= select max({{ time_col }}) from {{ this }} - interval '{{ time_in_days }} day'
```

Reply:
also gave me an error
```
2022-01-30 00:28:54.264179 (Thread-3): Database Error in model metric_miner_count (models/metrics/metric_miner_count.sql)
  001003 (42000): SQL compilation error:
  syntax error line 14 at position 27 unexpected 'select'.
  syntax error line 14 at position 38 unexpected 'block_timestamp'.
  syntax error line 17 at position 1 unexpected ','.
```
compiled dbt statement looks like this
```
      create or replace temporary table HARMONY.DEV.metric_gas_used_count__dbt_tmp  as
      (select * from(
            

with

daily as (

    select 
        date_trunc('day', block_timestamp) as metric_date,
        'daily' as metric_period,
        sum(gas_used) as gas_count
    from HARMONY.DEV.blocks
    where 
        block_timestamp >= select max(block_timestamp) from HARMONY.DEV.metric_gas_used_count - interval '3 day'
    
    group by 1

),

hourly as (

    select 
        date_trunc('hour', block_timestamp) as metric_date,
        'hourly' as metric_period,
        sum(gas_used) as gas_count
    from HARMONY.DEV.blocks
    where 
        block_timestamp >= select max(block_timestamp) from HARMONY.DEV.metric_gas_used_count - interval '3 day'
    
    group by 1

),

minute as (

    select 
        date_trunc('minute', block_timestamp) as metric_date,
        'minute' as metric_period,
        sum(gas_used) as gas_count
    from HARMONY.DEV.blocks
    where 
        block_timestamp >= select max(block_timestamp) from HARMONY.DEV.metric_gas_used_count - interval '3 day'
    
    group by 1

),

final as (
    select * from daily
    union all
    select * from hourly
    union all
    select * from minute
)

select * from final
            ) order by (metric_date)
      );
      
```